### DISCUSSION TITLE
[Feature] Display images from all shared albums

### DISCUSSION THREAD

**Discussion by lichenscon** [damongolding/immich-kiosk#95] [Feature request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I would like to display images from all albums that are shared with the api key's user. By that way we can easily decide which images should be shown. 
For example my girlfriend and I could share albums of holidays with a separate user and remove old holidays from the slideshow the same way.

**Comment by jschwalbe** [comment]
Clarifying question: so you would have three users on your Immich server, and you and the gf share with user 3? Then any image/album shared with user 3 gets displayed? Interesting idea.

**Comment by lichenscon** [comment]
Exactly. As this user is only for the Kiosk, it does not collect any images. As I like to collect images from holidays etc. in albums, it would be the easiest to only have to share these.

I do not need or want just informational images like where I parked my car or images from work to be shown in the slideshow.

**Comment by damongolding** [comment]
I was thinking about something the other day that might work in this situation.

I was thinking of adding the special keyword `all` to the `album` and `person` option. If a user wanted to show all their albums it would be a pain to add each ID, more so if they have lots of albums/people.

In this situation you would use the 3rd accounts details and the `all` keyword. Then if you wanted to add/remove an album from rotation you would either share/un-share said album?

**Reply by jschwalbe** [reply]
That's a pretty slick way to do it.
I never thought about this, but the API key is tied to a specific user, right?

**Reply by damongolding** [reply]
> That's a pretty slick way to do it.
> 
> I never thought about this, but the API key is tied to a specific user, right?

Yeah, so in this instance. The api key would be the one generated from the 3rd account.

**Reply by lichenscon** [reply]
That is exactly what I thought I would be the way to go.

**Reply by chris-burrow-apps** [reply]
I was thinking exactly this yesterday! Would make it perfect for me. I have 3 users (me, partner, shared) so when we want to share event albums, we share with the 'shared' account so it can be shown on a google home.

**Reply by damongolding** [reply]
Are you guys willing to be beta testers as I have new build with `album=shared` that would achieve this?

**Reply by lichenscon** [reply]
I would love to! What information would you need?

Edit:
After a short test I can confirm `<URL>?album=shared` is working! I will test it the next days!

**Reply by damongolding** [reply]
> I would love to! What information would you need?

I assume you're running Kiosk via Docker? I will build a dev image tomorrow (my tomorrow) and give out the image link if that's ok?

**Reply by damongolding** [reply]
> I would love to! What information would you need?
> 
> 
> 
> Edit:
> 
> After a short test I can confirm `<URL>?album=shared` is working! I will test it the next days!

Now I am thoroughly confused, that is unless you've cloned the source code of the branch?

**Comment by damongolding** [comment]
If anyone is interested in being a beta tester they can use the docker image `damongolding/immich-kiosk-development:0.11.0` to test out the `album=shared` and `album=all` features (please not that shared albums to not show when using all). If you do so and have any feedback you can either leave it here or hit me up on the Immich Discord server. There is a little documentation at the bottom of the album section [here](https://github.com/damongolding/immich-kiosk/tree/task/release?tab=readme-ov-file#albums)

**Reply by chris-burrow-apps** [reply]
I have just tested album=shared with the dev version. Works exactly you have described. No longer do I need to list every shared album on Immich. Thanks for building the feature 👍

**Reply by chris-burrow-apps** [reply]
Feedback: 
Is there any way to have maybe 3 toggles: 

- album=shared (only shared albums)
- album=user (only albums user has created) 
- album=all (both shared and created albums)
- album=favourites (favorite photos selected with heart)

**Reply by damongolding** [reply]
> Feedback: 
> 
> Is there any way to have maybe 3 toggles: 
> 
> 
> 
> - album=shared (only shared albums)
> 
> - album=user (only albums user has created) 
> 
> - album=all (both shared and created albums)
> 
> - album=favourites (favorite photos selected with heart)

Some interesting suggestions. I'm not sure how feasible they all are (limitations of the api possibly) but definitely food for thought. I've been meaning to allow favourites as a filter and adding that keyword might be the solution!

**Reply by lichenscon** [reply]
I have tested it for a good time now and I love it! The only thing I noticed is that you get error messages, when you remove an album from the sharing and the cache has not refreshed.
It also works to display own and shared images by adding both keywords to the album option.

Thank you very much for your work! Now I have a digital immich frame with an old Microsoft Surface!

**Reply by damongolding** [reply]
> I have tested it for a good time now and I love it! The only thing I noticed is that you get error messages, when you remove an album from the sharing and the cache has not refreshed.
> 
> It also works to display own and shared images by adding both keywords to the album option.
> 
> 
> 
> Thank you very much for your work! Now I have a digital immich frame with an old Microsoft Surface!


Excellent! You might also benefit from the new sleep mode feature that will also come with v0.11!

Out of curiosity are you running it on your surface in full screen? And if so, how are you enabling that?

**Reply by lichenscon** [reply]
Currently my setup is the following:
I have an Surface Pro 5 (2017) running Windows 10 with a passwordless account with autologin. In the autostart folder, I have a linkfile with the following content:

`C:\Path\to\librewolf.exe --kiosk https://immich-kiosk.my-domain`

This opens my Librewolf browser in kiosk mode. The only issue is that I have to trick a little bit to exit to the system.
On the Surface itself I can swipe to the left at the right border of the screen to open the settings and click the icon to open the settings app.
When I am on it via RustDesk, I can press CTRL+W to close the tab. I really like to change the settings on a server instead of the client with the immich-frame project, because I can do it remote, even if I give a device to my parents or my grandma.
In that case, I think the kiosk mode is even better than running it via the internal fullscreen option, because they cannot exit it.

**Reply by damongolding** [reply]
> I have tested it for a good time now and I love it! The only thing I noticed is that you get error messages, when you remove an album from the sharing and the cache has not refreshed.

I have added an icon to the menu in v0.11 that when clicked will flush the server cache and reload the device.
